assert/assertions.go

```diff
@@ -912,19 +912,25 @@ func containsElement(list interface{}, element interface{}) (ok, found bool) {
 		}
 	}()
 
-	if listKind == reflect.String {
+	switch listKind {
+	case reflect.String:
 		elementValue := reflect.ValueOf(element)
 		return true, strings.Contains(listValue.String(), elementValue.String())
-	}
-
-	if listKind == reflect.Map {
+	case reflect.Map:
 		mapKeys := listValue.MapKeys()
 		for i := 0; i < len(mapKeys); i++ {
 			if ObjectsAreEqual(mapKeys[i].Interface(), element) {
 				return true, true
 			}
 		}
 		return true, false
+	case reflect.Slice:
+		switch element := element.(type) {
+		case []byte:
+			if list, ok := list.([]byte); ok {
+				return true, bytes.Contains(list, element)
+			}
+		}
```

Comment:
Now about this

[]string{"foo", "bar"} contains "foo"

But what about []string{"foo", "bar"} containing 1, here 1 refer to the index 2 of the list, so "bar"

What is the expected return for Contains?

- an error about object type
- true
- false

Is it tested?

Reply:
I honestly found the Map behaviour testing keys a bit surprising, I'd find the array behaviour testing length even more surprising. Still worth adding the test via a new issue.